File: plugins/repository-search/src/main/java/org/jetbrains/idea/reposearch/DependencySearchProvider.java
package org.jetbrains.idea.reposearch;

import org.jetbrains.annotations.NotNull;

import java.util.List;
import java.util.concurrent.CompletableFuture;

public interface DependencySearchProvider {
  CompletableFuture<List<RepositoryArtifactData>> fulltextSearch(@NotNull String searchString);

  CompletableFuture<List<RepositoryArtifactData>> suggestPrefix(@NotNull String groupId, @NotNull String artifactId);

  boolean isLocal();
}

File: plugins/repository-search/common/src/main/java/org/jetbrains/idea/reposearch/DependencySearchProvider.kt
package org.jetbrains.idea.reposearch

import org.jetbrains.idea.maven.model.MavenRepoArtifactInfo

@Deprecated("Use DependencyCompletionContributor instead")
interface DependencySearchProvider {
  suspend fun fulltextSearch(searchString: String): List<MavenRepoArtifactInfo>

  suspend fun suggestPrefix(groupId: String, artifactId: String): List<MavenRepoArtifactInfo>

  fun isLocal(): Boolean

  val cacheKey: String
}
